File: backend/PropMate.Api/Models/ViewingSlot.cs
using System.Text.Json.Serialization;

namespace PropMate.Api.Models;

public class ViewingSlot
{
    public int Id { get; set; }
    public int PropertyId { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public bool IsAvailable { get; set; } = true;

    // Navigation property
    [JsonIgnore]
    public Property? Property { get; set; }
}

File: mobile/propmate_mobile/lib/models/viewing_slot.dart
class ViewingSlot {
  final int id;
  final int propertyId;
  final DateTime startTime;
  final DateTime endTime;
  final bool isAvailable;

  ViewingSlot({
    required this.id,
    required this.propertyId,
    required this.startTime,
    required this.endTime,
    required this.isAvailable,
  });

  factory ViewingSlot.fromJson(Map<String, dynamic> json) {
    return ViewingSlot(
      id: json['id'],
      propertyId: json['propertyId'],
      startTime: DateTime.parse(json['startTime']),
      endTime: DateTime.parse(json['endTime']),
      isAvailable: json['isAvailable'],
    );
  }
}
